source/features/conversation-navigation.tsx

```diff
@@ -0,0 +1,258 @@
+	}
+);
+
+async function getConversationList() {
+	const listQuery = getListQuery();
+	const list = await fetchConversationList(listQuery);
+	const currentNumber = getConversationNumber();
+	const currentConversation = list.find(item => item.number === currentNumber);
+	const {after, before} = getPageCursors(listQuery.page);
+
+	// `currentConversation` is from previous page, we need to fetch previous page items
+	if (currentConversation?.cursor === after) {
+		const newListQuery = {
+			...listQuery,
+			page: listQuery.page - 1
+		};
+		return {
+			list: await fetchConversationList(newListQuery),
+			listQuery: newListQuery
+		};
+	}
+
+	// `currentConversation` is from next page, we need to fetch next page items
+	if (currentConversation?.cursor === before) {
+		const newListQuery = {
+			...listQuery,
+			page: listQuery.page + 1
+		};
+		return {
+			list: await fetchConversationList(newListQuery),
+			listQuery: newListQuery
+		};
+	}
+
+	return {
+		list,
+		listQuery
+	};
+}
+
+function noListQuery(): boolean {
+	return !(
+		// Allowed also from global conversation list
+		pageDetect.isConversationList(new URL(document.referrer)) ||
+		(new URL(location.href)).searchParams.has('q')
+	);
+}
+
+function getListQuery() {
+	const referrerUrl = new URL(document.referrer);
+
+	// Coming from another issue with query in URL
+	if (!pageDetect.isConversationList(referrerUrl)) {
+		return parseConversationListURL(new URL(location.href));
+	}
+
+	// Default conversation lists, even global, don't have query in URL
+	// But it seems there is some refined-github feature that adds a default query in URL based on query input
+	// Make sure we have some query as a fallback based on current GitHub defaults
+	const query = referrerUrl.searchParams.get('q');
+
+	if (pageDetect.isGlobalConversationList(referrerUrl)) {
+		// Is global pull requests list
+		if (referrerUrl.pathname.split('/', 2)[1] === 'pulls') {
+			referrerUrl.searchParams.set('q', query ?? `is:open is:pr author:${getUsername()} archived:false`);
+		}
+
+		// Is global issues list
+		if (referrerUrl.pathname.split('/', 2)[1] === 'issues') {
+			referrerUrl.searchParams.set('q', query ?? `is:open is:issue author:${getUsername()} archived:false`);
+		}
+	}
+
+	if (pageDetect.isRepoConversationList(referrerUrl)) {
+		if (pageDetect.isRepoIssueList(referrerUrl)) {
+			referrerUrl.searchParams.set('q', `${query ?? 'is:issue is:open'} repo:${getRepoURL()}`);
+		}
+
+		if (pageDetect.isRepoPRList(referrerUrl)) {
+			referrerUrl.searchParams.set('q', `${query ?? 'is:pr is:open'} repo:${getRepoURL()}`);
+		}
+	}
+
+	return parseConversationListURL(referrerUrl);
+}
+
+function getConversationSearchQuery(first: number, after: string, query: string): string {
+	return `
+search(
+	type: ISSUE,
+	first: ${first},
+	after: "${after}",
+	query: "${query}"
+) {
+	edges {
+		cursor
+		node {
+			... on Issue { number, url }
+			... on PullRequest { number, url }
```

Comment:
I think:
```suggestion
			... on IssueOrPullRequest { number, url }
```

Reply:
Also I think that if you write it like this, you won't have to write `edges.node.map()` later, but just `node.map()`:

```gql
edges {
  cursor
}
node {
  ... on IssueOrPullRequest { number, url }
}
```

Reply:
It seems this is not valid. Also I need cursors to be near data.

![image](https://user-images.githubusercontent.com/1881266/90438037-12240980-e0dc-11ea-8373-520fd4ae63b2.png)

Reply:
No, that part is actually valid; it's the `IssueOrPullRequest` that's missing in this case sadly. This works:


```gql
{
  search(type: ISSUE, first: 10, query: "is:issue repo:sindresorhus/refined-github") {
    edges {
      cursor
    }
    nodes {
      ... on Issue {
        number
      }
      ... on PullRequest {
        number
      }
    }
  }
}
```

Reply:
In this case I get cursors separated from nodes in a `edges` array